### PR TITLE
platform specific getNowExpression() is not usable.

### PR DESCRIPTION
**Pull request by julien-sebire**
Changed platform proxy to use platform specific date formatting function instead of getNowExpression.
We can't use $this->dbalPlatform->getNowExpression() because sqlite, used in the tests for instance, returns `datetime('now')` which can not be parsed as a regular date.
We instead generate a date with php and then format it with $this->dbalPlatform->getDateTimeTzFormatString(), to still have the correct format to be inserted in db depending on the platform.

### CODE REVIEW

File: manifest.php [hunk 1]
```diff
@@ -31,7 +31,7 @@
     'label' => 'Generis Core',
     'description' => 'Core extension, provide the low level framework and an API to manage ontologies',
     'license' => 'GPL-2.0',
-    'version' => '11.4.3',
+    'version' => '11.4.5',
```
Comment (peetya) [inline manifest.php]: Why not `11.4.4`?

File: common/persistence/sql/class.Platform.php [hunk 2]
```diff
@@ -146,16 +146,41 @@ public function getMigrateSchemaSql($fromSchema,$toSchema){
...
-       return $date;
+        // We can't use $this->dbalPlatform->getNowExpression() because sqlite,
+        // at least used for the tests, returns `datetime('now')` which can
+        // not be parsed as a regular date.
+        // We instead generate a date with php and the format it with
+        // $this->dbalPlatform->getDateTimeTzFormatString(), to still have the
+        // correct format to be inserted in db.
+
+        $datetime = new \DateTime('now', new \DateTimeZone('UTC'));
+        $date = $datetime->format($this->dbalPlatform->getDateTimeTzFormatString());
```
Comment (peetya) [inline common/persistence/sql/class.Platform.php]: How about `return $datetime->...;`?

Comment (julien-sebire) [reply common/persistence/sql/class.Platform.php]: Fixed.

File: manifest.php (hunk 1 above)
Comment (julien-sebire) [reply manifest.php]: Previously existing tag not merged...

File: common/persistence/sql/class.Platform.php [hunk 3]
```diff
@@ -146,16 +146,39 @@ public function getMigrateSchemaSql($fromSchema,$toSchema){
...
+
+        $datetime = new \DateTime('now', new \DateTimeZone('UTC'));
+        return $datetime->format($this->dbalPlatform->getDateTimeTzFormatString());
+    }
+
+    /**
+     * Returns platform specific date formatting to store datetime field.
+     * @return string
+     */
+    public function getDateTimeFormatString()
```
Comment (bugalot) [inline common/persistence/sql/class.Platform.php]: Is it in use? If not, let's not add additional code ;)

**Comment by julien-sebire** [comment]
Added unit test for platform getNowExpression using platform specific datetime format.

**Pull request merged**